ports/espressif/bindings/espidf/__init__.c

```diff
@@ -129,6 +143,8 @@ STATIC const mp_rom_map_elem_t espidf_module_globals_table[] = {
 
     { MP_ROM_QSTR(MP_QSTR_erase_nvs), MP_ROM_PTR(&espidf_erase_nvs_obj)},
 
+    { MP_ROM_QSTR(MP_QSTR_get_total_psram), MP_ROM_PTR(&espidf_get_total_psram_obj)},
+    { MP_ROM_QSTR(MP_QSTR_get_reserved_psram), MP_ROM_PTR(&espidf_get_reserved_psram_obj)},
     { MP_ROM_QSTR(MP_QSTR_IDFError), MP_ROM_PTR(&mp_type_espidf_IDFError) },
     { MP_ROM_QSTR(MP_QSTR_MemoryError),      MP_ROM_PTR(&mp_type_espidf_MemoryError) },
```

Comment:
I think it would be OK to use existing exception classes here, like `SystemError` and `MemoryError`.

Reply:
@tannewt do you remember why you (I think?) used subclasses here instead of re-using the normal error? Happy to change it, but only if there's agreement about it.

Reply:
looks like scott added MemoryError (#3326) and I added IDFError (#3733)

Reply:
I made a new memory error subclass so that we could distinguish between running out of heap memory from running out of IDF memory if we wanted to.